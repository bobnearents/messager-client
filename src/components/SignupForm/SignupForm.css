.signup-form {
  background:#CFDBD5;
  height:90vh;
  color:#264653;
}

.signup-form form {
  text-align: center;
  width:15%;
  min-width:170px;
  display:flex;
  flex-direction: row;
  flex-flow: column;
  margin:auto;
}

.signup-form form button {
  margin:auto;
}

.signup-form input {
  margin-bottom:15px;
}

.form-header {
  margin-bottom:50px;
  margin-top:0;
  padding-top:50px;
  font-size: 3em;
}

.signup-form div {
  margin-top:25px;
}

.signup-form a {
  margin-top:5px;
  display:block;
  text-align: center;
  color:#264653;
}

.signup-form a:hover {
  text-decoration: none;
}

.error {
  color:red;
  font-size: 1.3em;
  margin-bottom:20px;
}
