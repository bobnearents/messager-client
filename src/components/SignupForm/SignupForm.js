import React, { Component } from "react";
import { Link } from "react-router-dom";
import ApiService from "../../services/api-service";
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import "./SignupForm.css";
import MessageContext from "../../context/message-context";

class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  static contextType = MessageContext;

  onSignupSuccess = (username, password) => {
    AuthApiService.postLogin({
      username: username,
      password: password
    })
      .then(res => {
        TokenService.saveAuthToken(res.authToken);
        this.context.setUser(username)
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  handleSubmit = ev => {
    ev.preventDefault();
    let { full_name, username, password, nickname } = ev.target;
    full_name = full_name.value;
    username = username.value;
    password = password.value;
    nickname = nickname.value;

    ApiService.createUser(
      full_name,
      username,
      password,
      nickname,
      () => this.onSignupSuccess(username, password)
    )
    .catch(res => {
      this.setState({error: res.error})
    })
  };

  render() {
    
    const error = this.state.error;
    return (
      <div className="signup-form">
        <h2 className="form-header">Sign Up!</h2>
        
        <form onSubmit={ev => this.handleSubmit(ev)}>
          {error && <span className = 'error'>{error}</span>}
          <label htmlFor="username">Username: </label>
          <input type="text" id="username" />

          <label htmlFor="password">Password: </label>
          <input type="password" id="password" />

          <label htmlFor="full_name">Full Name: </label>
          <input type="text" id="full_name" />

          <label htmlFor="Nickname">nickname: </label>
          <input type="text" id="nickname" />

          <button type="submit">Submit</button>

          <div>
            Already have an account? <Link to="/login">Log in here</Link>
          </div>
        </form>
      </div>
    );
  }
}

export default SignupForm;
