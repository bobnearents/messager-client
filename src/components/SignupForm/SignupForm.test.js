import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Route, Link, MemoryRouter } from "react-router-dom";

import SignupForm from "./SignupForm";

describe("<SignupForm />", () => {
  it("Renders without crashing", () => {
    shallow(<SignupForm />);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SignupForm />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
