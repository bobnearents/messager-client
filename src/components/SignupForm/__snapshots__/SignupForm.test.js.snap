// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupForm /> renders correctly 1`] = `
<div
  className="signup-form"
>
  <h2
    className="form-header"
  >
    Sign Up!
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      Username: 
    </label>
    <input
      id="username"
      type="text"
    />
    <label
      htmlFor="password"
    >
      Password: 
    </label>
    <input
      id="password"
      type="password"
    />
    <label
      htmlFor="full_name"
    >
      Full Name: 
    </label>
    <input
      id="full_name"
      type="text"
    />
    <label
      htmlFor="Nickname"
    >
      nickname: 
    </label>
    <input
      id="nickname"
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
    <div>
      Already have an account? 
      <a
        href="/login"
        onClick={[Function]}
      >
        Log in here
      </a>
    </div>
  </form>
</div>
`;
